channel2/impl.go

```diff
@@ -205,6 +215,13 @@ func (channel *channelImpl) SendWithPriority(m *Message, p Priority) (err error)
 	return nil
 }
 
+func (channel *channelImpl) SendBufferSafeWithPriority(m *Message, p Priority) (err error) {
+	copyBuf := make([]byte, len(m.Body))
+	copy(copyBuf, m.Body)
+	m.Body = copyBuf
+	return channel.SendWithPriority(m, p)
+}
+
```

Comment:
Considering a lot of the memory-handling code in `channel2` is less than optimal, why not just make it the default for all of the `SendXXX` methods to copy the inbound buffer, rather than adding this signature?

Reply:
It's only used in one spot. In most other places, including in the link and xgress send code, we don't need it. The only place we need it is in the sdk send code. I figured better to leave the places that don't need it with the more efficient form, and then we can go back and get rid of this when we add pooling to channel2.

Reply:
If that's the case you could move your `copy` code to the caller and leave the `channel2` interface intact.

Reply:
That's a good point. I put the copy buffer logic in the sdk. I'll remove the other method from foundation in my next PR.